### PR TITLE
Add GitHub Workflow Replacement for Jenkins job_PreCommit_Spotless

### PR DESCRIPTION
**Pull request by aleksandr-dudko**
The following jenkins jobs have been migrated:

https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Spotless_Commit/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Spotless_Cron/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Spotless_Phrase/

How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

by creating pull requests with changes in trigger paths
by adding a comment to the PR
by making a push

image

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/beam_PreCommit_Spotless.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  beam_PreCommit_Spotless:
```
Comment (magicgoody) [inline .github/workflows/beam_PreCommit_Spotless.yml:65]: could you please add timeout-minutes attribute

File: .github/workflows/beam_PreCommit_Spotless.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        uses: actions/setup-java@v3.8.0
+        with:
+          distribution: 'zulu'
+          java-version: '8'
+          cache: 'gradle'
+          check-latest: true
+      - name: run Spotless PreCommit script
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        with:
+          gradle-command: :sdks:java:core:jmh:spotlessCheck :sdks:java:core:jmh:checkStyleMain :sdks:java:core:jmh:checkStyleTest
```
Comment (andreydevyatkin) [inline .github/workflows/beam_PreCommit_Spotless.yml:99]: The result report step should also be added

**Review by andreydevyatkin — CHANGES_REQUESTED** [review]

File: .github/workflows/beam_PreCommit_Spotless.yml (hunk 2 above)
Comment (aleksandr-dudko) [reply .github/workflows/beam_PreCommit_Spotless.yml:99]: it's done.

**Review by andreydevyatkin — APPROVED** [review]
LGTM

File: .github/workflows/beam_PreCommit_Spotless.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
```
Comment (Abacn) [inline .github/workflows/beam_PreCommit_Spotless.yml:99]: thi is not correct. It just run against `:sdks:java:core:jmh` module. Should run on all java modules

File: .github/workflows/beam_PreCommit_Spotless.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    name: beam_PreCommit_Spotless (Run Spotless PreCommit)
+    timeout-minutes: 120
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Spotless PreCommit'
+    steps:
+      - name: Git checkout
```
Comment (Abacn) [inline .github/workflows/beam_PreCommit_Spotless.yml:76]: We now consolidated these first 2 steps into `./.github/actions/setup-action`:

e.g.

https://github.com/apache/beam/blob/7e365d78c797c4cd8b933857c780b5d681e597fd/.github/workflows/beam_PreCommit_Java_Examples_Dataflow_Java11.yml#L81

File: .github/workflows/beam_PreCommit_Spotless.yml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  statuses: read
+
+jobs:
+  beam_PreCommit_Spotless:
+    name: beam_PreCommit_Spotless (Run Spotless PreCommit)
+    timeout-minutes: 120
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' ||
```
Comment (Abacn) [inline .github/workflows/beam_PreCommit_Spotless.yml:72]: also, add a condition `github.event_name == 'workflow_dispatch' ||` here

Comment (aleksandr-dudko) [reply .github/workflows/beam_PreCommit_Spotless.yml:73]: It's done

**Review by Abacn — APPROVED** [review]
LGTM

**Pull request merged**